javascript/src/execution/scenario-execution-state.ts

```diff
@@ -63,10 +53,17 @@ export class ScenarioExecutionState implements ScenarioExecutionStateLike {
     }
   }
 
-  addMessages(messages: CoreMessage[], fromAgentIdx?: number): void {
-    for (const message of messages) {
-      this.addMessage(message, fromAgentIdx);
-    }
+  appendMessage(role: CoreMessage["role"], content: string): void {
+    const message: CoreMessage = { role, content } as CoreMessage;
+    this._messages.push({ ...message, id: generateMessageId() });
+  }
+
+  appendUserMessage(content: string): void {
+    this.appendMessage("user", content);
+  }
+
+  appendAssistantMessage(content: string): void {
+    this.appendMessage("assistant", content);
```

Comment:
hmmm it's super weird to have both addMessage and appendMessage right? also we don't want to expose those to the user, check out how the python code does it, it has a reference to the executor, and the messages actually live there, the ScenarioState is created every time we call a step instead, just as a nice helper to read the state, but not to set it, executor has the source of truth, so we don't need to make a lot of those setters public, risking the user to mess up with the simulation state

Reply:
ah this is left over, i missed it. removed!

Reply:
As for the other, I did a big bit of refactoring around something similar as I want to avoid a circle dependency, which we have in Python. Give me a mo.